src/Bicep.Core.Samples/Files/Functions/sys.json

```diff
@@ -1,478 +1,700 @@
+    "fixedParameters": [
+      {
+        "name": "base64Value",
+        "description": "The base64 representation to convert to a JSON object.",
+        "type": "string",
+        "required": true
+      }
     ],
     "minimumArgumentCount": 1,
     "maximumArgumentCount": 1,
     "flags": "default",
-    "typeSignature": "(param0: string): any",
+    "typeSignature": "(base64Value: string): any",
     "parameterTypeSignatures": [
-      "param0: string"
+      "base64Value: string"
     ]
   },
   {
     "name": "base64ToString",
-    "fixedParameterTypes": [
-      "string"
+    "description": "Converts a base64 representation to a string.",
+    "fixedParameters": [
+      {
+        "name": "base64Value",
+        "description": "The base64 representation to convert to a string.",
+        "type": "string",
+        "required": true
+      }
     ],
     "minimumArgumentCount": 1,
     "maximumArgumentCount": 1,
     "flags": "default",
-    "typeSignature": "(param0: string): string",
+    "typeSignature": "(base64Value: string): string",
     "parameterTypeSignatures": [
-      "param0: string"
+      "base64Value: string"
     ]
   },
   {
     "name": "bool",
-    "fixedParameterTypes": [
-      "any"
+    "description": "Converts the parameter to a boolean.",
+    "fixedParameters": [
+      {
+        "name": "value",
+        "description": "The value to convert to a boolean.",
+        "type": "any",
+        "required": true
+      }
     ],
     "minimumArgumentCount": 1,
     "maximumArgumentCount": 1,
     "flags": "default",
-    "typeSignature": "(param0: any): bool",
+    "typeSignature": "(value: any): bool",
     "parameterTypeSignatures": [
-      "param0: any"
+      "value: any"
     ]
   },
   {
     "name": "coalesce",
-    "fixedParameterTypes": [
-      "any"
-    ],
+    "description": "Returns first non-null value from the parameters. Empty strings, empty arrays, and empty objects are not null.",
+    "fixedParameters": [],
     "minimumArgumentCount": 1,
-    "variableParameterType": "any",
+    "variableParameter": {
+      "namePrefix": "arg",
+      "description": "The value to coalesce",
+      "type": "any",
+      "minimumCount": 1
+    },
     "flags": "default",
-    "typeSignature": "(param0: any, ... : any): any",
+    "typeSignature": "(... : any): any",
     "parameterTypeSignatures": [
-      "param0: any",
       "... : any"
     ]
   },
   {
     "name": "concat",
-    "fixedParameterTypes": [
-      "array"
-    ],
+    "description": "Combines multiple arrays and returns the concatenated array.",
+    "fixedParameters": [],
     "minimumArgumentCount": 1,
-    "variableParameterType": "array",
+    "variableParameter": {
+      "namePrefix": "arg",
+      "description": "The array for concatenation",
+      "type": "array",
+      "minimumCount": 1
+    },
     "flags": "default",
-    "typeSignature": "(param0: array, ... : array): array",
+    "typeSignature": "(... : array): array",
```

Comment:
>... [](start = 23, length = 3)

I've changed the modeling around variable params on some functions. Before we had 1 required and then variable params. Now it's all variable params with min count constraint. It will make it easier for me to generate variable argument names in signature help. (I don't have to keep track of starting index, basically.)